Code/GraphMol/Kekulize.cpp

```diff
@@ -512,17 +512,41 @@ void Kekulize(RWMol &mol, bool markAtomsBonds, unsigned int maxBackTracks) {
   //       - getBondType return aromatic
   // - all aromatic atoms return true for "getIsAromatic"
 
-  // first find the all the simple rings in the molecule
+  // first find the all the simple rings in the molecule that are not
+  // completely composed of dummy atoms
   VECT_INT_VECT arings;
-  if (mol.getRingInfo()->isInitialized()) {
-    arings = mol.getRingInfo()->atomRings();
+  boost::dynamic_bitset<> dummyAts(mol.getNumAtoms());
+  for (ROMol::AtomIterator atit = mol.beginAtoms(); atit != mol.endAtoms();
```

Comment:
Kekulize can be called at any point though, so I think having the test is necessary.

W.r.t. The dummy ats optimization: wouldn't that either require construction of a dynamic bitset of atoms that are in rings or a lot of searching. (Note: this isn't the place where a lot of time is spent anyway, so this may all be premature optimization).